### PR TITLE
Code development 

### PR DESCRIPTION
**Pull request by AndyZzzZzzZzz**
- Removed independent search script 
- Removed progress bar 
- Removed the verbose variable and corresponding print statements
- Removed whole graph search functions in raster_embedding
- Removed subgrid functions in raster_embedding

### CODE REVIEW

File: minorminer/utils/raster_embedding.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+        embs += sub_embs
+        if len(embs) >= max_number_of_embeddings:
+            break
+
+    # Get independent set of embeddings (removed, potential feature expansion)
+
+    embmat = np.asarray([[ie[v] for ie in embs] for v in sorted(subgraph.nodes)]).T
+
+    if verify_embeddings:
```
Comment (jackraymond) [inline minorminer/utils/raster_embedding.py]: We can move this part to tests/ (not required to approve pull request)

File: minorminer/utils/raster_embedding.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        for emb in embmat:
+            X = list(embedding.diagnose_embedding({p: [emb[p]] for p in range(len(emb))}, subgraph, A))
+            if X:
+                raise Exception("Embedding verification failed.")
+
+    assert len(np.unique(embmat)) == len(embmat.ravel()), "Embeddings are not unique."
```
Comment (jackraymond) [inline minorminer/utils/raster_embedding.py]: We will move this to tests/  (not required to approve pull request)

File: minorminer/utils/raster_embedding.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        B = _A.subgraph([f(_) for _ in tile]).copy()
+
+        sub_embs = search_for_subgraphs_in_subgrid(B, subgraph,
+                                                   max_number_of_embeddings=max_number_of_embeddings,
+                                                   **kwargs)
+        
+        for sub_emb in sub_embs:
+            _A.remove_nodes_from(sub_emb.values())
+
+        if verify_embeddings:
```
Comment (jackraymond) [inline minorminer/utils/raster_embedding.py]: We will move this to tests/  (not required to approve pull request)

File: minorminer/utils/subgraph.py [hunk 4]
```diff
@@ -0,0 +1,223 @@
+# Copyright 2023 D-Wave
```
Comment (jackraymond) [inline minorminer/utils/subgraph.py]: We can delete this module, looks like you successfully removed the dependency in raster_embedding

File: tests/utils/test_raster_embedding.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
+# test_raster_embedding.py
```
Comment (jackraymond) [inline tests/utils/test_raster_embedding.py:1]: Good placeholders! Start writing some tests when you are ready: maybe wait until my next pull request though (I'm going to simplify the signature of raster_embedding() once this pull request goes through.)

**Review by jackraymond — APPROVED** [review]
Just a shallow inspection, Kevin will do a slightly deeper dive.

File: minorminer/utils/raster_embedding.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+        max_number_of_embeddings (int, optional): Maximum number of embeddings to find. Defaults to np.inf.
+        verbose (bool, optional): Whether to print progress messages. Defaults to True.
+        **kwargs: Additional keyword arguments.
+
+    Returns:
+        list: A list of embeddings found.
+    """
+    embs = []
+    while True and len(embs) < max_number_of_embeddings:
+        temp = glasgow.find_subgraph(subgraph, B, timeout=timeout, triggered_restarts=True)
```
Comment (kevinchern) [inline minorminer/utils/raster_embedding.py:35]: I think we can expect to call `find_subgraph` with other parameters: should use `**kwargs` or explicitly enumerate the parameters instead. For reference, [find_subgraph](https://github.com/dwavesystems/minorminer/blob/main/minorminer/subgraph.pyx#L144)
(and for other occurrence of `find_subgraph`)

File: minorminer/utils/raster_embedding.py (hunk 2 above)
Comment (AndyZzzZzzZzz) [reply minorminer/utils/raster_embedding.py]: Should we still leave a try/catch block in the code?

File: minorminer/utils/raster_embedding.py [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+        topology (str, optional): The topology type ('chimera', 'pegasus', 'zephyr'). Defaults to 'pegasus'.
+        max_number_of_embeddings (int, optional): Maximum number of embeddings to find. Defaults to np.inf.
+        **kwargs: Additional keyword arguments.
+
+    Returns:
+        np.ndarray: An embedding matrix.
+    """
+    _A = A.copy()
+
+    assert list(subgraph.nodes) == list(range(len(subgraph))), "Subgraph must have consecutive nonnegative integer nodes."
```
Comment (kevinchern) [inline minorminer/utils/raster_embedding.py:66]: try/except

File: minorminer/utils/raster_embedding.py [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+def raster_embedding_search(
+        A, subgraph, raster_breadth=5,
+        topology='pegasus',
+        max_number_of_embeddings=np.inf,
+        **kwargs):
+    """
+    Searches for embeddings within a rastered subgraph.
+
+    Args:
+        _A (networkx.Graph): The hardware graph.
```
Comment (kevinchern) [inline minorminer/utils/raster_embedding.py:54]: docs need to be updated, but let's note it for when the code is closer to final

File: minorminer/utils/raster_embedding.py [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+        subgraph (networkx.Graph): The subgraph to embed.
+        raster_breadth (int, optional): Raster breadth. Defaults to 5.
+        topology (str, optional): The topology type ('chimera', 'pegasus', 'zephyr'). Defaults to 'pegasus'.
+        max_number_of_embeddings (int, optional): Maximum number of embeddings to find. Defaults to np.inf.
+        **kwargs: Additional keyword arguments.
+
+    Returns:
+        np.ndarray: An embedding matrix.
+    """
+    _A = A.copy()
```
Comment (kevinchern) [inline minorminer/utils/raster_embedding.py:64]: `A = A.copy()` instead

**Review by kevinchern — APPROVED** [review]

File: minorminer/utils/raster_embedding.py (hunk 2 above)
Comment (kevinchern) [reply minorminer/utils/raster_embedding.py]: oh good point, test or try/except?

Comment (kevinchern) [reply minorminer/utils/raster_embedding.py]: this shouldn't happen, so probably better a test.

**Pull request closed without merging**